### PR TITLE
Change tarball making procedure

### PR DESCRIPTION
**Pull request by levitte**
Since recently, OpenSSL tarballs are produced with 'make tar' rather
than 'make dist', as the latter has turned out to be more troublesome
than useful.

The next step to look at is why we would need to configure at all to
produce a Makefile just to produce a tarball.  After all, the tarball
should now only contain source files that are present even without
configuring.

Furthermore, the current method for producing tarballs is a bit
complex, and can be greatly simplified with the right tools.  Since we
have everything versioned with git, we might as well use the tool that
comes with it.

Added: util/mktar.sh, a simple script to produce OpenSSL tarballs.  It
takes the options --name to modify the prefix of the distribution, and
--tarfile tp modify the tarball file name specifically.

This also adds a few entries in .gitattributes to specify files that
should never end up in a distribution tarball.

-----

NOTE: this indirectly addresses #7691

### CODE REVIEW

**Review by mattcaswell — APPROVED** [review]

**Comment by levitte** [comment]
Merged.

master:
8d9535ec3e317641b8e551973c8cfe2ee1c89296 Remove all 'make dist' artifacts
8c209eeef426ded66ce99048f535f35d08b88462 Change tarball making procedure

1.1.1:
415c4a46695d32c47865c21d11a466f72d647179 Remove all 'make dist' artifacts
48c813ef3ac28363c355afe4b1594eaacf9886d4 Change tarball making procedure

1.1.0:
b97cbe05a3a4bf18b6bd6262dbda04b87256a54c Remove all 'make dist' artifacts
f68bfdf62805f1f3af4330754c7f141fa521b6dd Change tarball making procedure

File: util/mktar.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#! /bin/sh
+
```
Comment (richsalz) [inline util/mktar.sh]: copyright missing

Comment (levitte) [reply util/mktar.sh]: Good point

Comment (levitte) [reply util/mktar.sh]: #7696

File: Configurations/unix-Makefile.tmpl [hunk 2]
```diff
@@ -872,37 +872,8 @@ tags TAGS: FORCE
...
-	 $(PREPARE_CMD); \
-	 find . -type d -print | xargs chmod 755; \
-	 find . -type f -print | xargs chmod a+r; \
-	 find . -type f -perm -0100 -print | xargs chmod a+x); \
-	(cd $$TMPDIR; $(TAR_COMMAND) $$DISTDIR) \
-	| (cd $(SRCDIR); gzip --best > $(TARFILE).gz); \
-	rm -rf $$TMPDIR
-	cd $(SRCDIR); ls -l $(TARFILE).gz
-
-dist:
```
Comment (mspncp) [inline Configurations/unix-Makefile.tmpl]: Shouldn't the `dist` target remain and be aliased to `tar` in order not to break existing scripts?

```
dist:  tar
```

Comment (mspncp) [reply Configurations/unix-Makefile.tmpl]: Are you assuming that nowbody downstream is using the `dist` target? But even if this were the case, is it valid to remove a make target from a stable release?

Comment (levitte) [reply Configurations/unix-Makefile.tmpl]: I think we concluded that targets that we intend for ourselves only, such as release targets, are off limits to everyone else, or "use at your own risk".

**Pull request closed without merging**